ecosystem/typescript/sdk/src/providers/aptos_client.ts

```diff
@@ -672,7 +672,7 @@ export class AptosClient {
     }
     if (!(lastTxn as any)?.success) {
       throw new FailedTransactionError(
-        `Transaction ${txnHash} committed to the blockchain but execution failed`,
+        `Transaction ${txnHash} failed with an error: ${(lastTxn as any).vm_status}`,
```

Comment:
Yay :D Is there any way we can put the `vm_status` in a field in the `FailedTransactionError` class? I actually don't know if this makes sense, but it would make it easier to use instead of parsing it. Is it generally not a good idea to add fields to existing classes if the sdk exposes them with export, or would it not really matter

Reply:
we have the full txn object part of the `FailedTransactionError` class. but as mentioned in the PR comment, Error class is a bit weird as it only outputs the "message" prop, unless we wrap it with try/catch and then we can get access to the other class properties, it feels like a big change for this use case and that should be good for now.

We can always just return the full transaction object, but then we lost the "message" part, i.e you won't see `Transaction ${txnHash} failed with an error: ${(lastTxn as any).vm_status}` but only a txn object and then you would need to look for the `success` prop to see if it succeed or failed - basically same response for a success txn

Reply:
oh duh, I forgot the txn object is in the error! Yeah this is great, I think it makes most sense this way then